keyboards/redox/keymaps/tehl/keymap_rgb.c

```diff
@@ -0,0 +1,27 @@
+#include "keymap.h"
+#include "keymap_rgb.h"
+
+bool is_default_rgb_state = false;
+
+layer_state_t layer_state_set_rgb(layer_state_t state) {
+    switch (biton32(state)) {
+        case _ADMIN:
+            rgblight_sethsv_noeeprom_red();
+            rgblight_mode_noeeprom(RGBLIGHT_MODE_BREATHING + 2);
+            is_default_rgb_state = false;
+            break;
+        case _NUMPD:
+            rgblight_mode_noeeprom(RGBLIGHT_MODE_RAINBOW_MOOD + 1);
+            is_default_rgb_state = false;
+            break;
+        default:
+            // avoid restarting animation on enter/exit _RAISE
+            if (!is_default_rgb_state) {
+                rgblight_mode_noeeprom(RGBLIGHT_MODE_RAINBOW_SWIRL + 2);
+            }
+            is_default_rgb_state = true;
+            break;
+    }
+
+    return state;
+}
```

Comment:
This also should go in keymap.c.

Reply:
I wanted to keep some separation between the actual layout of the keys, and any additional logic which changes the behavior of the board. Is there a more appropriate way I can achieve this? Maybe moving the logic to userspace?

Reply:
Personally, I'm not a huge fan of this, but I understand the reasoning.

I think this should be okay, with the caveat that the `biton32` needs to be changed to `get_highest_layer`.